channeld/channeld.c

```diff
@@ -442,9 +449,12 @@ static void make_channel_local_active(struct peer *peer)
 						    annfeatures);
  	wire_sync_write(MASTER_FD, take(msg));
 
-	/* Tell gossipd and the other side what parameters we expect should
-	 * they route through us */
-	send_channel_update(peer, 0);
+	/* Under CI, because blocks come so fast, we often find that the
+	 * peer sends its first channel_update before the above message has
+	 * reached it. */
+	notleak(new_reltimer(&peer->timers, peer,
+			     time_from_sec(5),
+			     send_channel_initial_update, peer));
```

Comment:
I tried many things to fix this, including telling gossipd about the channel much earlier.  They were all ugly, and none foolproof.

Once connectd starts tracking channel_ids (which it has to to demux) it will probably be in a much better place to inform gossipd about the new channel, and avoid this race.